source/core/ib_stencils.cc

```diff
@@ -0,0 +1,272 @@
+//
+// Created by lucka on 2021-05-21.
+//
+
+#include <core/ib_stencils.h>
+
+
+
+template<int dim>
+unsigned int
+IBStencils<dim>::nb_points(unsigned int order)
+{
+    // The number of points used in the stencil excluding the DOF is equal to the order.
+    unsigned int nb_points=order;
+    // In the case where the cell is used directly a=to to find the solution at the IB only one point is needed.
+    if(order>4)
+        nb_points=1;
+    return nb_points;
+}
+
+template<int dim>
+std::vector<double>
+IBStencils<dim>::coefficients(unsigned int order)
+{
+    // The coefficients of the IB stencil assume a ratio of length between the distance of
+    // the farthest interpolation point and the DOF and the distance between the IB and the DOF of 1/8.
+    // The coefficient order goes from the coefficient of the DOF to the coefficient of the farthest interpolation point.
+    if (order==1) {
+        std::vector<double> coef(2);
+        coef[0] = 9;
+        coef[1] = -8;
+
+        return coef;
+    }
+    if (order==2) {
+        std::vector<double> coef(3);
+        coef[0]=153;;
+        coef[1]=-288;
+        coef[2]=136;
+
+        return coef;
+    }
+    if (order==3) {
+        std::vector<double> coef(4);
+        coef[0]=2925;
+        coef[1]=-8424;
+        coef[2]=8100;
+        coef[3]=-2600;
+
+        return coef;
+    }
+    if (order==4) {
+        std::vector<double> coef(5);
+        coef[0]=58905;
+        coef[1]=-228480;
+        coef[2]=332640;
+        coef[3]=-215424;
+        coef[4]=52360;
+
+        return coef;
+    }
+    if (order>4) {
+        // In this case the cell is directly used to find the solution at the IB position. In this case only one point is needed (the position of the point on the IB) and its coefficient is 1.
+        std::vector<double> coef(1);
+        coef[0] = 1;
+
+
+        return coef;
+    }
+
+    std::vector<double> coef(2);
+    coef[0] = 9;
+    coef[1] = -8;
+
+    return coef;
+}
+
+template<int dim>
```

Comment:
Maybe the .h would require a bit more explanation here?

Reply:
added some comments